src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,151 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.Ui;
+import duke.command.CommandKeyword;
+import duke.exception.DukeException;
+import duke.exception.IncompleteTaskDescriptionException;
+import duke.exception.InvalidCommandException;
+
+/**
+ * Contains the task list. It has operations to change the tasks in the list.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param tasksInfo Arraylist of tasks loaded from storage.
+     * @param ui An Ui instance to deal with interactions with the user.
+     */
+    public TaskList(ArrayList<String> tasksInfo, Ui ui) {
+        List<Task> list = tasksInfo.stream().map(Task::convertToTask).collect(Collectors.toList());
```

Comment:
Would this be a more descriptive variable name?
```suggestion
        List<Task> tasks = tasksInfo.stream().map(Task::convertToTask).collect(Collectors.toList());
```